clients/src/main/java/org/apache/kafka/clients/producer/MockProducer.java

```diff
@@ -255,4 +257,20 @@ public void complete(RuntimeException e) {
         }
     }
 
+    @Override
```

Comment:
I don't think we need to make these interface level changes in this patch. Those can come along with implementations later.. 

I am not sure whether we need the configs either -> those can come with each individual patch as needed.

What we do need is the RPC stubs.